libraries/radspreadprocessing/features/comments.md

```diff
@@ -0,0 +1,189 @@
+---
+title: Comments
+page_title: Comments 
+slug: radspreadprocessing-features-comments
+tags: comments, xlsx
+published: True
+position: 23
+---
+
+
+# Comments
+
+Since R1 2022 the SpreadProcessing library supports working with comments. Comments are used for marking information about a cell's data and can have one or multiple [Replies] (#replies). All comments can be found in the __Comments__ property of the worksheet, which is of the type __CommentsCollection__. This collection holds __SpreadsheetComment__ objects which represent the comments. Each comment has the following members:
+
+***Properties***:
+
+* __CellIndex:__  Gets or sets the cell index assigned to the comment. 
+* __Text:__ Gets or sets the text of the comment. 
+* __CreationDate:__ Gets or sets the date when the comment is created. Nullable.
+* __Author:__ Gets or sets the author assigned to the comment.
+* __Replies:__ Gets the comment replies. The list is sorted by CreationDate.
+* __IsResolved:__ Gets or sets a value indicating whether the comment is resolved.
+
+ >note All of the above properties will push a change to the undo stack when modified. 
+ 
+***Methods***:
+
+* __AddReply:__ Adds a SpreadsheetCommentReply to the ReplySortedCollection.The collection will be re-sorted by SpreadsheetReply`s CreationDate in ascending order after adding an object.
+* __RemoveReply:__ Removes the specified reply from the collection.
+
+## Working with the CommentsCollection
+
+### Adding comments
+
+To add a comment you need to specify the cell index to which the comment will be related, the author, the text content, and the creation date. Specifying the creation date is optional and by default, its value is the current date and time.
+
+#### __[C#] Example 1: Add comment__
+
+{{region cs-radspreadprocessing-features-comments_0}}
+	
+    CellIndex relatedCellIndex = new CellIndex(1, 1);
+    string author = "John Doe";
+    string text = "Comment Content";
+	DateTime creationDate = DateTime.Now;
+
+	worksheet.Comments.Add(relatedCellIndex, author, text, creationDate);
+
+{{endregion}}
+
+The above snippet will add a comment in cell B2.
+
+
+### Removing Comments
+
+To remove a comment specify the comment instance, which can be obtained from the CommentsCollection. 
```

Comment:
To remove a comment, you should specify the comment instance. This instance can be obtained from the CommentsCollection.